ucfcs/GrowthPlus
# Mack/parent portal changes

Closes issues #162, #161, and #119. These are some of the changes:

- A forgot pin screen has been fully implemented (you can get there from the parent login page)
- The parent signup screen has a spot to put in a phone number now
- ParentSchema and ParentSchemaService have a phone number field now
- There are no more toast messages: instead EditText boxes change to red when there are problems with the input
- The plus icon to add a child doesn't appear unless a parent has been created


To test:
- run the app
- go to the parent signup screen, test having incorrect input for the PIN and/or the phone number and make sure the right box(es) turn red
- go to the login page and test having bad input so the box turns red
- play around with the forgot pin screen in general, it has a pretty straightforward use

Note: the onResume method in ParentForgotPassword.java needs to be uncommented for the language translation to happen, to be honest I haven't looked into that at all
